### DISCUSSION TITLE
playerctl media issues

### DISCUSSION THREAD

**Discussion by Adforham** [i3/i3#6156] [Q&A] [answered]
if i have something like spotify and youtube opened, the way windows worked was when i pressed my play media key it played/paused the last media that i had paused/played, but playerctl defaulted to youtube even though i played spotify more recently, just because i had opened firefox before spotify, and i tried making a script for it but the script only played whichever media player came first in the array, since spotify was first in the array the script prioritized spotify, how do i make the script take into account which media player was most recently played, like if i just paused a youtube video and pressed my play media key, some random spotify song from 15 minutes ago shouldn't play, the video should, and vice versa, i eventually just gave up and resorted to asking you all, thanks in advance!

**Comment by KebabLord** [comment, marked as answer]
I don't understand your question or it's relation to i3, but anyways.
if you want to make the playerctl control the latest played media (it could be youtube, spotify etc.) you can use this approach:
```bash
# use this as a daemon to track name of the latest played media to a file, it will always run in the background
playerctl status --format '{{ playerName }}' --follow > /tmp/player-last

# when you want to pause/play the latest focused media, you can use it like this:
playerctl play-pause -p `tail -n 1 /tmp/player-last`
```
